homework/main.ipynb

```diff
@@ -908,21 +855,47 @@
     "# Define parameter grid for LSH\n",
     "param_grid = [\n",
     "    {'numHashTables': 3, 'bucketLength': 0.1},\n",
-    "    {'numHashTables': 5, 'bucketLength': 0.05},\n",
-    "    {'numHashTables': 7, 'bucketLength': 0.03},\n",
-    "    {'numHashTables': 10, 'bucketLength': 0.01}\n",
+    "    {'numHashTables': 3, 'bucketLength': 0.15},\n",
+    "    {'numHashTables': 3, 'bucketLength': 0.2},\n",
+    "    {'numHashTables': 4, 'bucketLength': 0.1},\n",
+    "    {'numHashTables': 4, 'bucketLength': 0.15},\n",
+    "    {'numHashTables': 4, 'bucketLength': 0.2},\n",
+    "    {'numHashTables': 5, 'bucketLength': 0.1},\n",
+    "    {'numHashTables': 5, 'bucketLength': 0.15},\n",
+    "    {'numHashTables': 5, 'bucketLength': 0.2}\n",
     "]\n",
     "\n",
+    "\n",
     "# Run grid search for parameter tuning\n",
     "results = grid_search_tuning(rescaledData, param_grid)\n",
     "\n",
     "# Print results\n",
     "for result in results:\n",
-    "    print(f\"Params: {result['params']}, Similar Pairs: {result['similar_pairs_count']}, Time: {result['computation_time']}s\")\n",
-    "\n",
-    "# Stop Spark session after completion\n",
-    "spark.stop()\n"
+    "    print(f\"Params: {result['params']}, Similar Pairs: {result['similar_pairs_count']}, Time: {result['computation_time']}s\")"
```

Comment:
Similar pairs is somewhat different from the accuracy measure. Let's discuss it in our chat and I will switch here to accuracy